udata/core/dataset/models.py

```diff
@@ -638,6 +638,14 @@ def clean(self):
         if self.frequency in LEGACY_FREQUENCIES:
             self.frequency = LEGACY_FREQUENCIES[self.frequency]
 
+        resources_ids = set()
```

Comment:
These manual checks only care about the current dataset though?

Reply:
Yes we cannot check for all resources in different dataset (but I think resources are always scoped by the dataset ID so it shouldn't be a problem, no?)

Reply:
As discussed, I think there are some rare routes where it is the case, but let's deal with these later on.